core/controlplane/config/tls_config.go

```diff
@@ -133,6 +133,7 @@ func (c *Cluster) NewTLSAssets(caKey *rsa.PrivateKey, caCert *x509.Certificate)
 		DNSNames: []string{
 			fmt.Sprintf("*.%s.compute.internal", c.Region),
 			"*.ec2.internal",
+			fmt.Sprintf("*.%s.compute.amazonaws.com", c.Region),
```

Comment:
Intended to be used only when `etcd.nodeIdentitiyProvider: EIP` because it relies on EC2 public hostnames stabilized with EIPs